modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/PageRedirect/PageRedirect.tsx

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import React from 'react';
+import {Outlet} from 'react-router-dom';
+
+import i18n from '../../i18n';
+import EmptyState from '../EmptyState';
+
+enum PageRedirectType {
+	Error = 'error',
+	Loading = 'loading',
+}
+
+export type PageRedirectProps = {
+	errorMessage?: string;
+	errorTitle: string;
+	errorType?: any;
+	outletMutate?: any;
+	pageToRender: string;
+};
+
+const PageRedirect: React.FC<PageRedirectProps> = ({
+	errorMessage,
+	errorTitle,
+	errorType,
+	outletMutate,
+	pageToRender,
```

Comment:
@alvarolealdl we can dry this component a little bit:

1. there's 3 `error` properties here, `message`, `tittle`, `type`, better not, we just need to put `error` here, and then (since is derived from `Error` `TestrayError`) we can check other parameters from here, something like `error.message`, `error.cause`, etc.
2. pageToRender -> This can be replaced by Children (Since it's the component we want to render, let's simplify)
3. outletMutate this should be passed during the `children` component
4. Add a state called `loading` and render the `<Loading />`  Component

The Component can be called `PageLayout` or `PageRenderer`.

Not necessary this is a "Redirect" action, since you maintain in the same page

Reply:
Makes total sense. Thanks @kevenleone